spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+describe Api::V1::BooksController, type: :controller do
+  include_context 'with Authenticated User'
+
+  describe 'GET #index' do
+    context 'when fetching all the books' do
+      let!(:books) { create_list(:book, 3) }
+
+      before do
+        get :index
+      end
+
+      it 'reponses with books json' do
+        expected = ActiveModel::Serializer::CollectionSerializer.new(
+          books, each_serializer: BookSerializer
+        ).to_json
+        expect(response.body.to_json) =~ JSON.parse(expected)
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when fetching a book' do
+      let!(:book) { create(:book) }
+
+      before do
+        get :show, params: { id: book.id }
```

Comment:
also test, when you pass a book id that does not exist or is invalid

Reply:
Done